src/modules/75-cf/pages/feature-flags/__tests__/FeatureFlagsPage.test.tsx

```diff
@@ -353,76 +345,89 @@ describe('FeatureFlagsPage', () => {
+        expect(useGetAllFeaturesMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            queryParams: expect.objectContaining({ tags: mockTagsPayload.tags[0].identifier })
+          })
+        )
+      })
     })
 
     test('it should be disabled if tags are loading', async () => {
-      mockImport('services/cf', {
-        useGetAllTags: () => ({ data: null, refetch: jest.fn(), loading: true })
-      })
+      useGetAllTagsMock.mockReturnValue({ data: undefined, refetch: jest.fn(), loading: true } as any)
 
       renderComponent()
 
-      expect(screen.getByPlaceholderText('- tagsLabel -')).toBeDisabled()
+      expect(
+        document.querySelector('[class="bp3-popover-wrapper MultiSelectDropDown--main MultiSelectDropDown--disabled"]')
+      ).toBeInTheDocument()
     })
 
     test('it should be disabled if it fails to fetch tags', async () => {
-      mockImport('services/cf', {
-        useGetAllTags: () => ({
-          data: null,
-          refetch: jest.fn(),
-          loading: false,
-          error: 'ERROR FETCHING TAGS'
-        })
-      })
+      useGetAllTagsMock.mockReturnValue({
+        data: undefined,
+        refetch: jest.fn(),
+        loading: false,
+        error: 'ERROR FETCHING TAGS'
+      } as any)
 
       renderComponent()
 
-      expect(screen.getByPlaceholderText('- tagsLabel -')).toBeDisabled()
+      expect(
+        document.querySelector('[class="bp3-popover-wrapper MultiSelectDropDown--main MultiSelectDropDown--disabled"]')
+      ).toBeInTheDocument()
     })
 
     test('it should call the tags endpoint when a tag is being searched', async () => {
-      const useGetAllTagsMock = jest
-        .spyOn(cfServices, 'useGetAllTags')
-        .mockReturnValue({ data: mockTagsPayload, loading: false, error: null, refetch: jest.fn() } as any)
+      useGetAllTagsMock.mockReturnValue({
+        data: mockTagsPayload,
+        loading: false,
+        error: null,
+        refetch: jest.fn()
+      } as any)
 
       renderComponent()
 
-      const SEARCHED_TAG = 'tag1'
+      await userEvent.click(screen.getByText('tagsLabel'))
 
-      await userEvent.type(await screen.findByPlaceholderText('- tagsLabel -'), SEARCHED_TAG)
+      const searchedTag = 'tag1'
 
-      await waitFor(() => {
+      await userEvent.type(screen.getAllByRole('searchbox')[1], searchedTag)
+
+      await waitFor(() =>
         expect(useGetAllTagsMock).toHaveBeenCalledWith(
-          expect.objectContaining({ queryParams: expect.objectContaining({ tagIdentifierFilter: SEARCHED_TAG }) })
+          expect.objectContaining({ queryParams: expect.objectContaining({ tagIdentifierFilter: searchedTag }) })
         )
-      })
+      )
+
+      await waitFor(() => expect(screen.getByText(searchedTag)).toBeInTheDocument())
+
+      for (let i = 1; i < mockTagsPayload.tags.length; i++) {
+        await waitFor(() => expect(screen.queryByText(mockTagsPayload.tags[i].name)).not.toBeInTheDocument())
+      }
```

Comment:
Not an issue, but a `for of` works well for things like this
```suggestion
      for (const { name } of mockTagsPayload.tags) {
        await waitFor(() => expect(screen.queryByText(name)).not.toBeInTheDocument())
      }
```

Let's you do destructuring as well.